### ISSUE TITLE
Loader when compiling Spell

### ISSUE THREAD

**Issue by burn83** [bpmn-io/bpmn-js-examples#200]
When I try to compile code like as example i receive an error that suggest me to add another loader to read correctly a row of the Spellprops.js in which we can retrieve "return <"
Instruction. Can, anyone helps me?

Spellprops.js in "parts" return an error

- Chrome 69, Node 10
- OS: Windows 11

**Comment by nikku** [comment]
Could you specify the exact project you're trying?

**Comment by burn83** [comment]
Yes. sure.

i tried to realize this one: 
https://github.com/bpmn-io/bpmn-js-examples/tree/master/properties-panel-extension

but my compiler seems to need another loader for compile that row of code. It seems that i must to configure Babel aproppriately. It's correct? There is another solution more simple?

**Comment by nikku** [comment]
Indeed. The project uses JSX and you need to add the babel-loader to transpile it to plain javascript: https://medium.com/@sivaraj-v/basic-webpack-4-and-es5-to-es6-transpiler-using-babel-dc66e72c86c6.

**Comment by nikku** [comment]
You do, as the error message indicates. At least it seems to be used under the hood.

What you need to enable is transpilation of JSX (inside an Angular project). Not sure how that works. Alternatively you'd need to split-up the extension (to be transpiled individually) and the integration into your Angular app.

**Comment by burn83** [comment]
"Alternatively you'd need to split-up the extension (to be transpiled individually) and the integration into your Angular app. ..."
How can do this?

**Comment by nikku** [comment]
> > "Alternatively you'd need to split-up the extension (to be transpiled individually) and the integration into your Angular app. ..."
>
> How can do this?

:one: You create your customizations in a fork of https://github.com/bpmn-io/bpmn-js-examples/tree/master/properties-panel-extension. That project compiles JSX -> JS.

:two: Consume the transpiled JS resources produced in :one: in your Angular app.

To get additional feedback on this case I'd ask you to create a topic in [our forum](https://forum.bpmn.io/) and cross-reference this stack. Maybe users already solved this issue in a simpler manner.

**Comment by burn83** [comment]
" That project compiles JSX -> JS." .. in my fork, after creation of it, i see an IDE in a browser that reproduce exactly my Visual Studio IDE. So, the files are the same :( .... Can you help me?

**Comment by burn83** [comment]
So i try this one ... https://babeljs.io/repl ... in order to convert a part that includes jsx logic. I obtained a pure javascript but browser, in console, tell me ... React is not defined..

**Comment by burn83** [comment]
OK ... i found a complete solution to convert SpellProps.js(x) in SpellProps.js, in pure mode.
`"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

exports["default"] = _default;

var _propertiesPanel = require("@bpmn-io/properties-panel");
var _bpmnJsPropertiesPanel = require("bpmn-js-properties-panel");
var _react = require("react");

function _default(element) {
  return [{
    id: 'spell',
    element: element,
    component: Spell,
    isEdited: _propertiesPanel.isTextFieldEntryEdited
  }];
}

function Spell(props) {
  var element = props.element,
    id = props.id;
  var modeling = (0, _bpmnJsPropertiesPanel.useService)('modeling');
  var translate = (0, _bpmnJsPropertiesPanel.useService)('translate');
  var debounce = (0, _bpmnJsPropertiesPanel.useService)('debounceInput');
  
  var getValue = function getValue() {
    return element.businessObject.spell || '';
  };
  
  var setValue = function setValue(value) {
    return modeling.updateProperties(element, {
      spell: value
    });
  };

  return /*#__PURE__*/_react.createElement(_propertiesPanel.TextFieldEntry, {
    id: id,
    element: element,
    description: translate('Apply a black magic spell'),
    label: translate('Spell'),
    getValue: getValue,
    setValue: setValue,
    debounce: debounce
  });
}
`

and this is the output...

Now in browser, on the console, i find this error : " Uncaught (in promise): TypeError: Cannot add property __, object is not extensible" ... It's a bug?! or ... what?

**Comment by burn83** [comment]
![image](https://user-images.githubusercontent.com/17144371/201930947-5eaac1c8-2fc3-4086-841c-3712be9392e4.png)

can anyone help me?

**Comment by nikku** [comment]
@burn83 This looks like another Angular <-> bpmn.io tooling incompatibility issue.

Please share a [codesandbox](https://codesandbox.io/) that allows us to reproduce this behavior.

We aim for compatibility with existing UI frameworks. Experience has shown that some (Angular, [Vue](https://vuejs.org/api/reactivity-advanced.html#markraw)) do interesting magic though that you may need to disable for bpmn-js to work reliably.

**Comment by nikku** [comment]
We plan to simplify our examples to not use JSX/transpilation anymore (https://github.com/bpmn-io/bpmn-js-examples/issues/206). That should workaround most of the complications you mentioned in this issue.

**Comment by minzhenyu** [comment]
in my peoject, vue2， it doesn't work
![image](https://user-images.githubusercontent.com/12856051/223088422-23b9ce16-5bf0-4abf-bb80-c76ff45cc001.png)
